ignite/cmd/testnet_multi_node.go

```diff
@@ -0,0 +1,163 @@
+                        bonded: 100000000stake
+                        - name: validator1
+                        bonded: 100000000stake
+                        - name: validator2
+                        bonded: 200000000stake
+                        - name: validator3
+                        bonded: 300000000stake
+
+
+			The "multi-node" command allows developers to easily set up, initialize, and manage multiple nodes for a 
+			testnet environment. This command provides full flexibility in enabling or disabling each node as desired, 
+			making it a powerful tool for simulating a multi-node blockchain network during development.
+
+			Usage:
+					ignite testnet multi-node [flags]
+
+		
+
+		`,
+		Args: cobra.NoArgs,
+		RunE: testnetMultiNodeHandler,
+	}
+	flagSetPath(c)
+	flagSetClearCache(c)
+	c.Flags().AddFlagSet(flagSetHome())
+	c.Flags().AddFlagSet(flagSetCheckDependencies())
+	c.Flags().AddFlagSet(flagSetSkipProto())
+	c.Flags().AddFlagSet(flagSetVerbose())
+	c.Flags().BoolP(flagResetOnce, "r", false, "reset the app state once on init")
+	c.Flags().String(flagNodeDirPrefix, "validator", "prefix of dir node")
+
+	c.Flags().Bool(flagQuitOnFail, false, "quit program if the app fails to start")
+	return c
+}
+
+func testnetMultiNodeHandler(cmd *cobra.Command, _ []string) error {
+	session := cliui.New(
+		cliui.WithVerbosity(getVerbosity(cmd)),
+	)
+	defer session.End()
+
+	return testnetMultiNode(cmd, session)
+}
+
+func testnetMultiNode(cmd *cobra.Command, session *cliui.Session) error {
+	chainOption := []chain.Option{
+		chain.WithOutputer(session),
+		chain.CollectEvents(session.EventBus()),
+		chain.CheckCosmosSDKVersion(),
+	}
+
+	if flagGetCheckDependencies(cmd) {
+		chainOption = append(chainOption, chain.CheckDependencies())
+	}
+
+	// check if custom config is defined
+	config, _ := cmd.Flags().GetString(flagConfig)
+	if config != "" {
+		chainOption = append(chainOption, chain.ConfigFile(config))
+	}
+
+	c, err := chain.NewWithHomeFlags(cmd, chainOption...)
+	if err != nil {
+		return err
+	}
+
+	cfg, err := c.Config()
+	if err != nil {
+		return err
+	}
+
+	numVal, amountDetails, err := getValidatorAmountStake(cfg.Validators)
+	if err != nil {
+		return err
+	}
+	nodeDirPrefix, _ := cmd.Flags().GetString(flagNodeDirPrefix)
+
+	outputDir, err := xfilepath.Join(igcfg.DirPath, xfilepath.Path("local-chains/"+c.Name()+"d/"+"testnet/"))()
+	if err != nil {
+		return err
+	}
+	args := chain.MultiNodeArgs{
+		OutputDir:             outputDir,
+		NumValidator:          strconv.Itoa(numVal),
+		ValidatorsStakeAmount: amountDetails,
+		NodeDirPrefix:         nodeDirPrefix,
+	}
+
+	resetOnce, _ := cmd.Flags().GetBool(flagResetOnce)
+	if resetOnce {
+		// If resetOnce is true, the app state will be reset by deleting the output directory.
+		err := os.RemoveAll(outputDir)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = c.TestnetMultiNode(cmd.Context(), args)
+	if err != nil {
+		return err
+	}
+
+	time.Sleep(2 * time.Second)
```

Comment:
why the sleep?

Reply:
initializing the nodes directory will take a small amount of time, I left 2s as an estimate for initializing 9 nodes max